File: frontend/src/types/audit.ts
// Domain types mirroring backend models. One source of truth for UI.

export type Category =
  | "legitimacy"
  | "financial"
  | "online_presence"
  | "compliance"
  | "social_proof"
  | "growth";

export const CATEGORIES: Category[] = [
  "legitimacy",
  "financial",
  "online_presence",
  "compliance",
  "social_proof",
  "growth",
];

export const CATEGORY_LABEL: Record<Category, string> = {
  legitimacy: "Legitimacy",
  financial: "Financial",
  online_presence: "Online Presence",
  compliance: "Compliance",
  social_proof: "Social Proof",
  growth: "Growth",
};

export type RiskRating = "green" | "amber" | "red";

export interface CategoryScore {
  score: number;
  justification: string;
}

export interface Discrepancy {
  category: Category;
  claim: string;
  evidence: string;
  severity: "low" | "med" | "high";
}

export interface FrameworkBundle {
  swot: Record<string, string[]>;
  pestle: Record<string, string[]>;
  porters: Record<string, string>;
  risk_heatmap: Record<string, number>;
  benchmark_kpis: Array<Record<string, string>>;
}

export interface CategoryEvidence {
  category: Category;
  summary_markdown: string;
  sources: string[];
  raw: Record<string, unknown>;
  error: string | null;
}

export interface AuditReport {
  company_name: string;
  url: string;
  overall_risk: RiskRating;
  overall_summary: string;
  scores: Record<Category, CategoryScore>;
  discrepancies: Discrepancy[];
  frameworks: FrameworkBundle;
  evidence: Record<Category, CategoryEvidence>;
}

export type StreamEvent =
  | { type: "started"; url: string }
  | { type: "extractor"; category: Category; phase: "started" | "completed" | "failed"; detail: string }
  | { type: "report"; data: AuditReport }
  | { type: "error"; message: string };

File: backend/app/models/audit.py
"""Domain models for the audit pipeline."""
from __future__ import annotations

from enum import Enum
from typing import Any

from pydantic import BaseModel, Field, HttpUrl


class Category(str, Enum):
    LEGITIMACY = "legitimacy"
    FINANCIAL = "financial"
    ONLINE_PRESENCE = "online_presence"
    COMPLIANCE = "compliance"
    SOCIAL_PROOF = "social_proof"
    GROWTH = "growth"


class RiskRating(str, Enum):
    GREEN = "green"
    AMBER = "amber"
    RED = "red"


class AuditRequest(BaseModel):
    url: HttpUrl
    company_name: str | None = None


class CategoryEvidence(BaseModel):
    """Raw, structured evidence collected for a single category."""

    category: Category
    summary_markdown: str
    sources: list[str] = Field(default_factory=list)
    raw: dict[str, Any] = Field(default_factory=dict)
    error: str | None = None


class CategoryScore(BaseModel):
    score: int = Field(ge=0, le=10)
    justification: str


class Discrepancy(BaseModel):
    category: Category
    claim: str
    evidence: str
    severity: str = "med"


class FrameworkBundle(BaseModel):
    swot: dict[str, list[str]]
    pestle: dict[str, list[str]]
    porters: dict[str, str]
    risk_heatmap: dict[str, int]
    benchmark_kpis: list[dict[str, Any]]


class AuditReport(BaseModel):
    company_name: str
    url: str
    overall_risk: RiskRating
    overall_summary: str
    scores: dict[Category, CategoryScore]
    discrepancies: list[Discrepancy]
    frameworks: FrameworkBundle
    evidence: dict[Category, CategoryEvidence]
